Assets/Scripts/ObjectPool/ObjectPool.cs

```diff
@@ -0,0 +1,56 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class ObjectPool : MonoBehaviour
+{
+    [SerializeField] private int _spawnAmount;
+    [SerializeField] private GameObject _object;
+
+    private GameObject _parent;
+
+    private Queue<GameObject> _objectPool = new Queue<GameObject>();
```

Comment:
You're forcing us to work with GameObjects. What if I want to make an objectpool of a struct? Maybe a class? Is this impossible?